premake5.lua

```diff
@@ -84,6 +84,40 @@
 	solution "Premake5"
 		configurations { "Release", "Debug" }
 		location ( _OPTIONS["to"] )
+		
+		configuration "Debug"
+			defines     "_DEBUG"
+			flags       { "Symbols" }
+
+		configuration "Release"
+			defines     "NDEBUG"
+			flags       { "OptimizeSize" }
+
+		configuration "vs*"
+			defines     { "_CRT_SECURE_NO_WARNINGS" }
+
+		configuration "vs2005"
+			defines   {"_CRT_SECURE_NO_DEPRECATE" }
+			
+		configuration "cygwin"
+			--gccprefix "i686-pc-cygwin-"
+			defines   { "LUA_USE_POSIX" }
+
+		configuration "linux or bsd or hurd"
+			defines     { "LUA_USE_POSIX", "LUA_USE_DLOPEN" }
+
+		configuration "macosx"
+			defines     { "LUA_USE_MACOSX" }
+
+		configuration { "macosx", "gmake" }
+			toolset "clang"
+			buildoptions { "-mmacosx-version-min=10.4" }
+			linkoptions  { "-mmacosx-version-min=10.4" }
+
+		configuration "aix"
+			defines     { "LUA_USE_POSIX", "LUA_USE_DLOPEN" }
+
+		configuration {}
```

Comment:
There's another patch ongoing that adds a binary module and requires the same LUA settings as premake and links to pramke5.
I've moved all non-exe related things out of premake project as other project should also get the same (LUA) config parameters.
see: https://groups.google.com/d/msg/premake-development/QGnDJ_m1elI/SB5xcGPo4doJ